Mobile Buy SDK/Mobile Buy SDK/Payment Providers/BUYWebCheckoutPaymentProvider.m

```
@@ -0,0 +1,193 @@
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import "BUYWebCheckoutPaymentProvider.h"
+#import "BUYCheckout.h"
+#import "BUYClient+Checkout.h"
+
+@import SafariServices;
+
+NSString * BUYSafariCallbackURLNotification = @"kBUYSafariCallbackURLNotification";
+NSString * BUYURLKey = @"url";
+NSString * const BUYWebPaymentProviderId = @"BUYWebPaymentProviderId";
+
+static NSString *const WebCheckoutCustomerAccessToken = @"customer_access_token";
+
+@interface BUYWebCheckoutPaymentProvider () <SFSafariViewControllerDelegate>
+@property (nonatomic, strong) BUYCheckout *checkout;
+@property (nonatomic, strong) BUYClient *client;
+@end
+
+@implementation BUYWebCheckoutPaymentProvider
+
+@synthesize delegate;
+
+- (instancetype)initWithClient:(BUYClient *)client
+{
+	NSParameterAssert(client);
+	
+	self = [super init];
+	
+	if (self) {
+		_client = client;
+	}
+	
+	return self;
+}
+
+- (BOOL)isInProgress
+{
+	return (self.checkout != nil);
+}
+
+- (NSUInteger)hash
+{
+	return self.identifier.hash;
+}
+
+- (BOOL)isEqual:(id)object
+{
+	return ([object isKindOfClass:[self class]] && [self.identifier isEqual:[object identifier]]);
+}
+
+- (NSString *)identifier
+{
+	return BUYWebPaymentProviderId;
+}
+
+- (void)startCheckout:(BUYCheckout *)checkout
+{	
+	if (self.isInProgress) {
+		return;
```

We should throw an exception, actually.